### ISSUE TITLE
The 'math' documentation is incomplete

### ISSUE THREAD

**Issue by ChrisJefferson** [jekyll/jekyll#4146]
jekyll/site/_docs/extras.md talks about how kramdown supports mathjax, and how to write maths. However, the advice there doesn't seem complete, I think it needs something like:

You also need to explicitly include mathjax library in your web pages. You can read all about how to integrate mathjax at the [mathjax website](http://docs.mathjax.org/en/latest/start.html), or just add the following to  `_includes/head.html`, inside the `head` tag.

```
<script type="text/javascript"
            src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
```

**Comment by envygeeks** [comment]
It's only our job to give you an idea, it's Kramdowns job to explain that idea in detail.  Please file the ticket upstream and ask them to provide more accurate documentation if doesn't already include it.

**Comment by ChrisJefferson** [comment]
That seems like weird advice --  why would Kramdown tell you how to configure jekyll?

**Comment by envygeeks** [comment]
What you suggested has nothing to do with configuring Jekyll, it has to do with you using a feature of Kramdown after we told you that you *could* use that feature, now it's Kramdowns job to explain to you how to use that feature.  By the logic you just applied we should be storing all of Kramdowns documentation on our site, along with Redcarpet, Pgyments and every dependency we use.

The word before "supports" is who the ticket belongs to.

**Comment by envygeeks** [comment]
Let me rephrase it, if Maths support requires an extra library for it to work with Kramdown, it's their job to tell you that.

**Comment by ChrisJefferson** [comment]
I can see you aren't going to shift on this, but I'm confused by your comment:

> The word before "supports" is who the ticket belongs to.

From the page I reference, the only usage of the word supports is: "There are a number of (optional) extra features that Jekyll supports ..."

I can't imagine why you think that Kramdown is going to tell users where to add script tags to jekyll (or their website in general), but I suppose I (like everyone else) will just google for jekyll and mathjax and find a webpage that tells me how to fix it.

**Comment by envygeeks** [comment]
I won't budge on a lot of things and this isn't one of them.  My resistance is never a finality... a revision in my thinking happens in the face of a strong argument.

My supports was in reference to your first comment "kramdown supports" -- that said the reason I am resistant is because this isn't about Jekyll, this is about all the other sites too. You are using a Kramdown feature, yes, sure we did mention it because we want you to know the cool stuff you can do with Jekyll but, if you can't search "kramdown mathjax" and get an immediate result on the Kramdown site saying "add this library to your website for it to work" then it's a bug in Kramdown because it's their job to tell you how to get Maths to work with *their* library and if they don't then we are only fixing the problem for Jekyll and that's fixing a bug on OS X only when the bug also exists on Linux and Windows.

Now, on our end, I do think we should link off to the Kramdown page about Maths if they indeed have one (and they do: http://kramdown.gettalong.org/math_engine/mathjax.html) rather than saying "you need this and this library to get it to work" because that might not always be the case and now because we carry documentation that Kramdown themselves should have we have to re-alter our documentation.

**Comment by ChrisJefferson** [comment]
Not sure why you mentioned 'OS X', just checking to be clear that we are on the same issue.

Kramdown runs fine, it generates html correctly with 'mathjax'-style comments included. There is no need to change anything with regards the running of Kramdown.

The problem is that we need to make sure the 'mathjax' javascript library is loaded in the webpages we generate which include Kramdown mathjax output, else the resulting pages don't work (as there is no magic javascript).

The page you link to ( http://kramdown.gettalong.org/math_engine/mathjax.html ) would be a sensible place for mathjax to chose to give more explicit guidance on how to add mathjax (or link to mathjax's "how to include" page)

**Comment by envygeeks** [comment]
I can't tell if you are trying to troll or not because I have been overly clear on what I mean and that OS X comment was a clear analogy.  I've said all I need to say. This is distracting me from fixing bugs on other parts of Jekyll.

**Comment by envygeeks** [comment]
And looking at that page, we already do explicitly mention mathjax and link to it to imply you need it.

**Comment by ChrisJefferson** [comment]
I have pushed a change to the upstream page you mentioned. Once that is fixed, maybe jekyll can be fixed.

Be it won't be by me. Calling me a troll was nasty and upsetting. I won't be communicating with you, or working with jekyll on github again. Too mean a place for me.

**Comment by envygeeks** [comment]
I'm sorry you feel that way.  Hope you have a good day.

**Comment by parkr** [comment]
So if I understand what happened here, Chris said "hey, could you mention in your documentation that you need to include this JavaScript to get MathJax working" and Jordon said "no, that's not our job, the kramdown documentation for mathjax should do that."

Maybe the best way forward is for us to link out to the MathJax info in the kramdown documentation. What do you think?

P.S. @ChrisJefferson, sorry you didn't have a nice experience here. Jordon said "I can't tell if you are trying to troll or not," which isn't calling you a troll. We get a lot of people who are insistent about things, which sometimes feels like trolling when we have a couple dozen issues to go through. We could be better about communicating what we understand from the issuer and why we think we should do it differently. Sorry that this issue upset you; we're all human here, and we have no bad intentions. Just a misunderstanding. Sorry again!

**Comment by ChrisJefferson** [comment]
@park, @envygeeks : To be honest I was in a bad mood when I arrived here, mainly because I'd spent about 45 minutes trying to figure out why all my maths was invisible, but no errors were occurring (which is the current behaviour of out-of-the-box jekyll), and I do feel the current stuff is confusing -- it was unclear (to me) it's my job to add mathjax to my pages.

If you google you'll find a whole bunch of blog entries describing that you have to explicitly include mathjax to get jekyll to render maths. The kramdown documentation is bad in this area, perhaps just add a comment to make clear the user must get mathjax:

Note you must manually add the [mathjax](http://docs.mathjax.org/en/latest/start.html) library to any pages containing LaTeX.

**Comment by StevenBlack** [comment]
To be honest, merely documenting you *can* do something isn't enough.  An example should be provided.

Either do it right, or don't mention it at all.

**Comment by envygeeks** [comment]
@ChrisJefferson and that's the kind of thing that poses a good argument to me, if you had to spend 45 minutes and were angry, @StevenBlack black is right, we are doing it wrong, maybe we do need examples and also a link saying "you *must* manually add mathjax to get maths"

**Comment by parkr** [comment]
Room for improvement for sure. Hey @gjtorikian, would you happen to have any good resources we can link to or further ideas to improve this documentation? I know you have worked on math stuff in Jekyll before. :)

**Comment by rebornix** [comment]
From my perspective, it's not a simple JavaScript referencing issue, you can even replace *JavaScript* by *CSS* or *font file*. The reason why @ChrisJefferson can't figure out how to make mathjax right at first is that he may have no idea about how Jekyll themes/layouts work. This is common for newbies. 

I suggest to add some highlights in *Plugins/External Libraries* section, like 
> If the plugin/library/gem you install requires external JavaScript/CSS/Resources, remember to link them in layout files which are placed under `_layout` most of the time.
